### ISSUE TITLE
Itemization bullet affects text rendering

### ISSUE THREAD

**Issue by fbruetting** [UberWriter/uberwriter#135]
### ENVIROMENT
- Linux distribution: Fedora Silverblue 30
- Desktop Enviroment: Gnome
- DE version: 3.32
- GTK version: 3.24.7

### BUG

I think this is self-explaining:

![Bildschirmfoto von 2019-04-03 01-57-30](https://user-images.githubusercontent.com/10520767/55443813-59c3f700-55b4-11e9-9bb9-6c27ba690328.png)

**Comment by somas95** [comment]
This one is going to be painful 
We use regexp to do that (and it's far from being trivial) 
There are problems when using those symbols (*,_,etc) as text. For example, it's pretty typical to have something like !(some image)[foo_alice_bar.png] and it's screwed too

**Comment by fbruetting** [comment]
Can you maybe steal some code parts from e.g. Ghostwriter or use GtkTextView’s markup?

**Comment by goncalossilva** [comment]
At least Ghostwriter is also using regular expressions, which is also not ideal.

Over time the direction we should go is using Pandoc's `native` (or `json`) output, which provides us with an AST to work with. For your example:

```
[BulletList
 [[Plain [Str "Normal"]]
 ,[Plain [Emph [Str "Not",Space,Str "italic"]]]
 ,[Plain [Str "Italic\8230?!",Space,Str "*Not",Space,Str "italic\8230?!"]]]]
```

This is a non-trivial transition though, but I think we'll get there at some point.

**Comment by goncalossilva** [comment]
I've looked into this a little more and it's not possible to do accurately at the moment. The reason being that Pandoc's AST doesn't keep source mapping information, which would be needed to correctly map entities to their original location.

Tracked here: https://github.com/jgm/pandoc/issues/4565

I am unsure how long it will be before it lands, but @jgm is working on jgm/commonmark-hs which supposedly already contains all of this information and might.

**Comment by alerque** [comment]
Not a fix by any means, but one interesting note is that I usually don't run into this particular issue because I use Pandoc as a normalizer, feeding my Markdown through it at frequent intervals (even in *pre_commit* hooks) to guarantee my prose are safe to round-tripp into and out of Pandoc. Pandoc's default list marker is `-` not `*`, so list items end up using a symbol that is not so confusing to Regular Expressions. Likewise for underscores, which syntax highlighters tend to get wrong a lot too.

In the case of the example in this issue, in normalizes to:

```markdown
-   Normal
-   *Not Italic*
-   Italic...?! *Not Italic...?!*
```

...which probably not not run afoul of the RegEx weaknesses of the original.

**Comment by fbruetting** [comment]
And what does avoiding a bug help for fixing it?

**Comment by alerque** [comment]
@fbruetting It doesn't. But it might give end users (like you and me) a bit of frustration relief until it is fixed.

**Comment by goncalossilva** [comment]
I've played with this and was able to source map Pandoc's AST: https://github.com/goncalossilva/uberwriter/commit/fc65581524508c67974d0f28b97908ec957b1c64#diff-41a44a3a5ede2e154a603038d3e3dde7R27

It's not that complex considering the AST is mostly sorted. In theory, this nets us a 100% accurate interpretation of the tree.

The problem, and a big one, is that this is very slow. It easily goes over 1 second for large documents such as Pandoc's user guide. We can't possibly provide a solid composing experience when you have to wait a few seconds for the UI to provide feedback. Optimizing this is possible, but non-trivial.

We'll have to wait for Pandoc to provide this. Hopefully in the next major version. :crossed_fingers:

**Comment by somas95** [comment]
Great! Do yo know if there are plans to implement it in the next version? It would be really cool if Pandoc did that

**Comment by goncalossilva** [comment]
There's some insight from @jgm in the issue I linked before, namely that by integrating `commonmark-hs` this would be possible. I am unsure if this limits us to CommonMark, though.

**Comment by lavendthomas** [comment]
My pull request fixes this bug by needing a non-blank (non-space) character right after the last opening asterisk. Does that cover your use-case?  

@somas95 you mentionned `!(some image)[foo_alice_bar.png]` beeing screwed. Do you think it's reasonable to check for spaces around the underscores? I just checked and pandoc doesn't recognise underscores in the middle of a word to put the section of the word in emphasis. 

Do you have examples of something else that's broken, if we continue to use regexp for the time being, they will have to be handled one by one.

**Comment by somas95** [comment]
Solved by  #185
